### DISCUSSION TITLE
Unable to get assets included using AvaloniaResource from code

### DISCUSSION THREAD

**Discussion by Mrcubix** [AvaloniaUI/Avalonia#9073] [Q&A] [answered]
My project is based on the xplat template and the assets i need are in `/Assets/`, the template contain these lines in the project file:

```xml
<ItemGroup>    
  <AvaloniaResource Include="Assets\**" />
</ItemGroup>
```

These assets are available without any issues in controls.
However, i need to get a list of assets in code in order to display the appropriate image depending on the situation.

I looked at both of the pages below as well as a dozen other pages, all stating basically the same thing.

https://docs.avaloniaui.net/docs/getting-started/assets
https://github.com/AvaloniaUI/Avalonia/issues/4488

However, i can't figure out why i am still unable to get a list of assets in my code.

here is what i'm trying at the moment:

```cs
var assets = AvaloniaLocator.Current.GetService<IAssetLoader>();

 var assemblyName = Assembly.GetExecutingAssembly().GetName().Name; // Destiny-Activity-Track-Analyzer

var pathResmProject = new Uri($"resm:{assemblyName}/Assets");
var pathAvaresProject = new Uri($"avares://{assemblyName}/Assets");

var resmProject = assets.GetAssets(pathResmProject, null);
var averesProject = assets.GetAssets(pathAveresProject, null);

var ressourcesNames = Assembly.GetExecutingAssembly().GetManifestResourceNames(); // ["!AvaloniaResources"]

var pathResmRessources = new Uri($"resm:{ressourcesNames[0]}/Assets");
 var pathAvaresRessources = new Uri($"avares:/{ressourcesNames[0].Replace("!", "")}/Assets");

var resmRessources = assets.GetAssets(pathResmRessources, null);
var averesRessources = assets.GetAssets(pathAveresRessources, null);
```

`resmProject`, `averesProject`, `resmRessources` & `averesRessources` are all null at runtime.

At first, i thought the issue was related to the assembly name, so i tried the same chunk of code above in a new xplat template,
the result is unfortunately the same

Here is the content of `assets` at runtime 

![image](https://user-images.githubusercontent.com/39861216/193451907-f868dab8-ff70-4fb8-9f08-4326b337b487.png)
Here is the content of `AvaloniaRessources`, a private member of `assets`
![image](https://user-images.githubusercontent.com/39861216/193451934-902ce871-0041-4935-b4f4-8324e9f08e49.png)

Update: 

Seems like i'm able to obtain empty enumerables using the following
```cs
var icons = assets.GetAssets(new Uri($"resm:Tracker.Assets), null);
```

Any ideas?

**Comment by maxkatz6** [comment, marked as answer]
Just tried with simple sandbox project. Seems to work fine for me. Can you provide a reproducible project?

> Assembly.GetExecutingAssembly().GetName().Name

Note that executing assembly might be different from one where resources were defined.

>  var pathAvaresRessources = new Uri($"avares:/{ressourcesNames[0].Replace("!", "")}/Assets");

Small typo, should be "avares://". Though GetManifestResourceNames wouldn't work here anyway, as it's not an assembly name.

**Reply by Mrcubix** [reply]
Went back to my apartment today which mean i'm on another machine, and `pathAvaresProject` is now not null?
Only assumption i can make at this point since nothing changed in my local repo is that there are already some kind of issues with something, software wise. ~~OS install on my new machine~~.

Also tried it on a linux VM i have access to, it work properly there too.

> Note that executing assembly might be different from one where resources were defined.

I figured out that much from looking at the github issue i linked and the doc

>>var pathAvaresRessources = new Uri($"avares:/{ressourcesNames[0].Replace("!", "")}/Assets");
>
> Small typo, should be "avares://". Though GetManifestResourceNames wouldn't work here anyway, as it's not an assembly name.

my bad for the typo, as for why i tried this, it was probably just out of desperation and trying to make it work so i don't have to come up with a separate tool just to update the specific part the Icons will be used in whenever a new thing is added.

I'm sorry for making this discussion for an issue caused by a potentially faulty software ~~OS install~~

Update: It doesn't really make sense for it to be an OS issue at all, i'll try and figure out what the issue is on my main machine next week-end
